File: imagegame_ui/src/services/InAppPurchaseService.js
import http from "../http-common"; 

class InAppPurchaseService {
  getAllInAppPurchases(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/inAppPurchase/inAppPurchases`, searchDTO);
  }

  get(inAppPurchaseId) {
    return this.getRequest(`/inAppPurchase/${inAppPurchaseId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/inAppPurchase?field=${matchData}`, null);
  }

  addInAppPurchase(data) {
    return http.post("/inAppPurchase/addInAppPurchase", data);
  }

  update(data) {
  	return http.post("/inAppPurchase/updateInAppPurchase", data);
  }
  
  uploadImage(data,inAppPurchaseId) {
  	return http.postForm("/inAppPurchase/uploadImage/"+inAppPurchaseId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new InAppPurchaseService();

File: src/main/java/com/imagegame/service/InAppPurchaseService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.InAppPurchase;
import com.imagegame.dto.InAppPurchaseDTO;
import com.imagegame.dto.InAppPurchaseSearchDTO;
import com.imagegame.dto.InAppPurchasePageDTO;
import com.imagegame.dto.InAppPurchaseConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface InAppPurchaseService extends GenericService<InAppPurchase, Integer> {

	List<InAppPurchase> findAll();

	ResultDTO addInAppPurchase(InAppPurchaseDTO inAppPurchaseDTO, RequestDTO requestDTO);

	ResultDTO updateInAppPurchase(InAppPurchaseDTO inAppPurchaseDTO, RequestDTO requestDTO);

    Page<InAppPurchase> getAllInAppPurchases(Pageable pageable);

    Page<InAppPurchase> getAllInAppPurchases(Specification<InAppPurchase> spec, Pageable pageable);

	ResponseEntity<InAppPurchasePageDTO> getInAppPurchases(InAppPurchaseSearchDTO inAppPurchaseSearchDTO);
	
	List<InAppPurchaseDTO> convertInAppPurchasesToInAppPurchaseDTOs(List<InAppPurchase> inAppPurchases, InAppPurchaseConvertCriteriaDTO convertCriteria);

	InAppPurchaseDTO getInAppPurchaseDTOById(Integer inAppPurchaseId);







}





